File: Demo/Static/CPP/mainClass.cpp
#include <iostream>
#include "Example1\counter.h"
#include "Example2\mathutils.h"
#include "Example3\myclass.h"

using namespace std;

int main()
{
    Counter c1;
    Counter c2;
    Counter c3;

    cout << c3.getCount() << endl; // 3

    // if count not static the output is 1

    int mx = MathUtils::max(3, 4);

    cout << mx << endl; // 4

    MyClass m1;
    MyClass m2;
    MyClass m3;

    cout << m3.getCount() << endl;

    // error count is a private member
    // cout << MyClass::count << endl;

    m3.staticMehtod();

    MyClass::staticMehtod();

    return 0;
}

File: Demo/Static/PY/main_class.py
# 1. Static Data (Class Variables):
# Static data, also known as class variables, are variables defined at the class level.
# They are shared by all instances (objects) of the class.

# 2. Static Methods:
# They are typically used for utility functions that are related to the class
# but don't necessarily need to modify object state.


from Example1.count import Count
from Example2.math_helper import MathHelper
from Example3.my_class import MyClass


c1 = Count()
c2 = Count()
c3 = Count()

print(c3.get_count())


result = MathHelper.add(5, 3)
print(result)

result = MathHelper.multiply(5, 3)
print(result)


MyClass.static_method()


m1 = MyClass()
print(MyClass.count)


m2 = MyClass()
print(MyClass.get_count())
